### DISCUSSION TITLE
how to use onBeforeRouteLeave

### DISCUSSION THREAD

**Discussion by dbzx10299** [nuxt/nuxt#28303] [Questions]
In the docs for [`onBeforeRouteLeave`](https://nuxt.com/docs/api/utils/on-before-route-leave) there is no explanation and the vue router docs are a bit confusing. I tried to use it in a component and I get the error
```
[Vue Router warn]: No active route record was found when calling `onBeforeRouteLeave()`. 
Make sure you call this function inside a component child of <router-view>. 
Maybe you called it inside of App.vue?
```

I did try to use this in a script setup block
```html
<script setup>
onBeforeRouteLeave(/* ... */)
</script>
```

**Comment by xjccc** [comment]
Like this?

```
onBeforeRouteLeave((to, from, next) => {
   next()
})
```

**Reply by dbzx10299** [reply]
That‘s exactly what I did and it gives the above error I mentioned

**Reply by xjccc** [reply]
Could you provide a mini repo for this?
